sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -745,6 +825,80 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[Dataset]] by computing the given [[Column]] expression for each element.
+   *
+   * {{{
+   *   val ds = Seq(1, 2, 3).toDS()
+   *   val newDS = ds.select(expr("value + 1").as[Int])
+   * }}}
+   * @since 1.6.0
+   */
+  def select[U1: Encoder](c1: TypedColumn[T, U1]): Dataset[U1] = {
```

Comment:
I don't think this is ever called. `select(Column*)` will always be preferred:

https://gist.github.com/vlad17/93f1cb57aad42eb7de33f92d6282a44f